testApp/src/Components/Sidebar.svelte

```diff
@@ -0,0 +1,91 @@
+<!-- Sidebar.svelte -->
+<script lang="ts">
+    import { createEventDispatcher, onMount, onDestroy } from 'svelte';
+    export let sidebarOpen: boolean;
+    export let selectFranchise: (id: number) => void;
+
+    const dispatch = createEventDispatcher();
+
+    function handleSelectFranchise(id) {
+        selectFranchise(id);
+        sidebarOpen = !sidebarOpen;
+    }
+
+    function handleClickOutside(event) {
+        if (!event.target.closest('.sidebar, .fa-bars')) {
+            sidebarOpen = false;
+        }
+    }
+
+    function closeSidebar() {
+        sidebarOpen = !sidebarOpen;
+    }
+
+    onMount(() => {
+        document.addEventListener('click', handleClickOutside);
+    });
+
+    onDestroy(() => {
+        document.removeEventListener('click', handleClickOutside);
+    });
+</script>
+
+<div class="sidebar" class:open={sidebarOpen}>
+    <div class="icons">
+        <i class="fa-light fa-magnifying-glass nav-icon"></i>
+        <i class="fa-light fa-shopping-cart nav-icon"></i>
+        <i class="fal fa-user nav-icon"></i>
+        <i class="fal fa-xmark nav-icon" on:click={closeSidebar}></i>
+    </div>
+    <a href="#" on:click={() => handleSelectFranchise(17)}>Star Wars</a>
+    <a href="#" on:click={() => handleSelectFranchise(4)}>Marvel</a>
+</div>
+
+<style>
+    .sidebar {
+        height: 100%;
+        width: 250px;
+        position: fixed;
+        top: 0;
+        right: -250px;
+        background-color: white;
+        box-shadow: 0 12px 20px rgba(0, 0, 0, 0.1);
+        transition: right 0.3s ease;
+        padding-top: 60px;
+        z-index: 1000;
+        border-radius: 50px 0 0 50px;
+    }
+
+    .sidebar a {
+        padding: 12px 20px;
+        text-decoration: none;
+        font-size: 1.2rem;
+        font-weight: 600;
+        color: black;
+        display: block;
+        transition: background-color 0.3s ease;
+        font-family: Crimson-Text;
```

Comment:
When using custom fonts, it's always a good idea to provide a fallback for when the CDN is unavailable or if someone has disabled these fonts. A quick search tells me this is a `serif` font, you could add that to the definition. I also don't think I've seen any script tag anywhere to include this font, right?

Reply:
I decided to just use the fond in global.css everywhere i did provide a fallback there.

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-6a7ee04012624d3db7b781ffd397b00ae6dff2154c43fc5f20624ebc0fdb4b5a